cms/datasets/models.py

```diff
@@ -1,58 +1,167 @@
+        other code that may be using the same base queryset instance.
+
+        Args:
+            token: Bearer token for API authentication
+
+        Returns:
+            Cloned queryset with token attached
+        """
+        clone: ONSDatasetApiQuerySet = self.clone()
+        clone.token = token
+        return clone
+
     def get_results_from_response(self, response: Mapping) -> Iterable:
         results: Iterable = response["items"]
+
         return results
 
     def fetch_api_response(self, url: str | None = None, params: Mapping | None = None) -> dict:
-        api_response: dict = super().fetch_api_response(url=url, params=params)
+        # Add Authorization header if token is set
+        headers = dict(self.http_headers) if self.http_headers else {}
+        if self.token:
+            # Note: Don't prepend "Bearer " - it's already in the token
+            headers["Authorization"] = self.token
+
+        if url is None:
+            url = self.base_url
+        if params is None:
+            params = {}
+
+        is_detail_request = url.startswith(ONSDataset.Meta.detail_url.split("%s", maxsplit=1)[0])
+
+        try:
+            response = requests.get(url, params=params, headers=headers, timeout=30)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                logger.warning("Rate limit exceeded when fetching datasets", extra={"url": url})
+                raise
+            # Check for 5xx server errors (500-599)
+            if e.response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                logger.error(
+                    "Server error when fetching datasets",
+                    extra={"url": url, "status_code": e.response.status_code},
+                )
+                raise
+            raise
+        except requests.exceptions.RequestException as e:
+            logger.error("Request failed when fetching datasets", extra={"url": url, "error": str(e)})
+            raise
+
+        api_response: dict = response.json()
+
+        # api_response should be a dict, let's raise a clear error if not
+        if not isinstance(api_response, dict):
+            logger.error("Invalid API response format when fetching datasets", extra={"url": url, "params": params})
+            raise ValueError("Invalid API response format, expected a dictionary-like object")
+
+        if is_detail_request:
+            api_response = self._process_detail_response(api_response)
+
         # The dataset API returns the per page count as "count" and the total results as "total_count"
         # Queryish expects "count" to be the total results count, so override it here
         if count := api_response.get("total_count"):
             api_response["count"] = count
+
         return api_response
 
+    def _process_detail_response(self, response: dict) -> dict:
```

Comment:
Nit: Mark this as a static method?. Surprisingly, our linter doesn't pick this up.